app/src/main/java/kth/jjve/xfran/utils/WorkoutUtils.java

```diff
@@ -0,0 +1,86 @@
+package kth.jjve.xfran.utils;
+
+import android.util.JsonReader;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import kth.jjve.xfran.models.Workout;
+
+public class WorkoutUtils {
+
+    //string builder for exercises from workout obj
+    public static String buildExerciseString(Workout workout) {
+        StringBuilder exercises = new StringBuilder();
+        ArrayList<String> exercisesArray = workout.getDetails();
+        for (String s : exercisesArray) {
+            exercises.append(s).append("\n");
+        }
+        return exercises.toString();
+    }
+
+    /* ------------ METHODS FOR PARSING WORKOUT LIST ------------ */
+    public static void parseWorkouts(JsonReader reader, ArrayList<Workout> dataSet) throws IOException {
+        // read json array, loop through workouts on dataSet and read each of them
+        reader.beginArray();
+        while (reader.hasNext()) {
+            dataSet.add(readWorkout(reader));
+        }
+        reader.endArray();
+    }
+
+    public static Workout readWorkout(JsonReader reader) throws IOException {
+        // initialize workout parameters
+        String title = null;
```

Comment:
Actually I tried that, but I had to initialize them null otherwise I got a "variable not initialized error". I was quite annoyed at this because it does look very useless. I googled and got nothing better than this. If you can make it work can you show it to me next meeting? 😃

Reply:
hmm, never heard about it, but if thats the case we have to roll with it 🤷🏻‍♀️